internal/provider/serverless_cluster_resource.go

```diff
@@ -0,0 +1,730 @@
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+
+	var ok bool
+	if r.provider, ok = req.ProviderData.(*tidbcloudProvider); !ok {
+		resp.Diagnostics.AddError("Internal provider error",
+			fmt.Sprintf("Error in Configure: expected %T but got %T", tidbcloudProvider{}, req.ProviderData))
+	}
+}
+
+func (r *serverlessClusterResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "serverless cluster resource",
+		Attributes: map[string]schema.Attribute{
+			"cluster_id": schema.StringAttribute{
+				MarkdownDescription: "The ID of the cluster.",
+				Computed:            true,
+			},
+			"display_name": schema.StringAttribute{
+				MarkdownDescription: "The display name of the cluster.",
+				Required:            true,
+			},
+			"region": schema.SingleNestedAttribute{
+				MarkdownDescription: "The region of the cluster.",
+				Required:            true,
+				Attributes: map[string]schema.Attribute{
+					"name": schema.StringAttribute{
+						MarkdownDescription: "The unique name of the region.",
+						Required:            true,
+					},
+					"region_id": schema.StringAttribute{
+						MarkdownDescription: "The ID of the region.",
+						Computed:            true,
+					},
+					"cloud_provider": schema.StringAttribute{
+						MarkdownDescription: "The cloud provider of the region.",
+						Computed:            true,
+					},
+					"display_name": schema.StringAttribute{
+						MarkdownDescription: "The display name of the region.",
+						Computed:            true,
+					},
+				},
+			},
+			"spending_limit": schema.SingleNestedAttribute{
+				MarkdownDescription: "The spending limit of the cluster.",
+				Optional:            true,
+				Computed:            true,
+				PlanModifiers: []planmodifier.Object{
+					objectplanmodifier.UseStateForUnknown(),
+				},
+				Attributes: map[string]schema.Attribute{
+					"monthly": schema.Int64Attribute{
+						MarkdownDescription: "Maximum monthly spending limit in USD cents.",
+						Optional:            true,
+						Computed:            true,
+					},
+				},
+			},
+			"automated_backup_policy": schema.SingleNestedAttribute{
+				MarkdownDescription: "The automated backup policy of the cluster.",
+				Optional:            true,
+				Computed:            true,
+				PlanModifiers: []planmodifier.Object{
+					objectplanmodifier.UseStateForUnknown(),
+				},
+				Attributes: map[string]schema.Attribute{
+					"start_time": schema.StringAttribute{
+						MarkdownDescription: "The time of day when the automated backup will start.",
+						Optional:            true,
+					},
+					"retention_days": schema.Int64Attribute{
+						MarkdownDescription: "The number of days to retain automated backups.",
+						Optional:            true,
```

Comment:
retention_days can not be changed, so it should be Computed.
in case it can be changed in the future, we can set it both the Optional and Computed